### ISSUE TITLE
Layout properly taken into account only the second time I display nglview widget

### ISSUE THREAD

**Issue by yakutovicha** [arose/nglview#865]
I am using nglview 2.7.1 and I encountered the following problem:

![Screenshot_2020-01-07 Untitled1](https://user-images.githubusercontent.com/13118485/71881627-f1053e80-3132-11ea-97fc-f7dc6bf3634c.png)

any idea why the protein is centred properly only the second time I display the nglview widget? Is there a quick fix for that?

**Comment by yakutovicha** [comment]
This may help. If I do `view.n_components = 1` before displaying the widget for the first time - it works without a problem.

**Comment by hainm** [comment]
Hi, please use attribute to set the widget.

```
import nglview as nv

view = nv.demo()
view.layout.width = '50%'
view
```

**Comment by yakutovicha** [comment]
Thanks, @hainm, with setting attribute it works.

I was just wondering: is it expected behaviour? I had the impression that nglview's interface is very much in line with jyputer widgets, where both approaches do work.

**Comment by hainm** [comment]
>  is it expected behaviour? 

yes, that's expected behavior. The feature is not implemented yet. 
Currently a widget only watches for the changes in its `layout`'s attributes. not the `layout` change itself.

https://github.com/arose/nglview/blob/0ad4c4ed69cca28c117ff5ee177ad2f3c64051e0/nglview/widget.py#L271-L279

> I had the impression that nglview's interface is very much in line with jupyter widgets, where both approaches do work.

yeah, almost.

**Comment by hainm** [comment]
If you still want to use `view.layout = {'width': '50%'}`, please call `view.handle_resize()` explicitly after that.
```
view = nv.demo()
view.layout = {'width': '50%'}
view.handle_resize()
view
```

**Comment by yakutovicha** [comment]
Alright, thanks for the explanation. Nice package by the way!

**Comment by yakutovicha** [comment]
Just discovered another issue: if I use `view.layout.width = '50%' `, as you suggested, everything works fine until I use the fullscreen. If I click on the fullscreen button and then return back to the notebook - the widget's height changes:
before fullscreen:
![Screenshot_2020-01-07 Untitled1(1)](https://user-images.githubusercontent.com/13118485/71910968-2cbff880-3173-11ea-9156-b27230edf652.png)
after fullscreen:
![Screenshot_2020-01-07 Untitled1(2)](https://user-images.githubusercontent.com/13118485/71911071-5bd66a00-3173-11ea-93ba-b654a57d6b40.png)

**Comment by yakutovicha** [comment]
However, with this approach:
```
view = nv.demo()
view.layout = {'width': '50%'}
view.handle_resize()
view
```

Everything seem to work fine.

**Comment by hainm** [comment]
thanks for your report. yeah, there are still many minor issues here and there and we will keep this in mind. cheers.
